src/app/modules/user/user-list.page.tsx

```diff
@@ -1,15 +1,34 @@
+import { useQuery } from '@apollo/client';
 import { UserRow } from 'app/components/user-row';
-import { fakeData } from 'app/services/fake-data';
+import { USER_LIST_TITLES } from 'app/constants';
+import { UserListQuery } from 'app/services';
 import { Table } from 'atomic/mol.table';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+interface UserData {
+  id: string;
+  name: string;
+  email: string;
+}
 
 export function UserListPage() {
-  const tableTitles = ['Name', 'E-mail'];
+  const { loading, error, data } = useQuery(UserListQuery, {
+    variables: { offset: 0, limit: 20 },
+  });
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (error?.graphQLErrors[0].name == 'UnauthorizedError') {
+      navigate('/login');
+    }
+  }, [error]);
 
   return (
     <>
       <h1>Users list</h1>
-      <Table titles={tableTitles}>
-        {fakeData.userList.map(({ id, name, email }) => {
+      <Table isLoading={loading} titles={USER_LIST_TITLES}>
+        {data?.users?.nodes.map(({ id, name, email }: UserData) => {
```

Comment:
Você tratou um caso de erro específico (Unauthorized), e foi um tratamento bem feito. Porém o que acontece com a tela se o erro for de outro tipo?